### PR TITLE
Fix exit codes for inferentia jobs

### PR DESCRIPTION
**Pull request by miguelvr**
checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

File: test/apis/batch/inferentia/cortex_inf.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+- name: image-classifier-resnet50
+  kind: BatchAPI
+  handler:
+    type: python
+    path: handler.py
+    config:
+      model_path: s3://cortex-examples/pytorch/image-classifier-resnet50
```
Comment (RobertLucian) [inline test/apis/batch/inferentia/cortex_inf.yaml:7]: Add a README/comment explaining that the highest batch size can only be 1 due to how the Inferentia model was compiled.

**Review by RobertLucian — APPROVED** [review]
LGTM - just a few comments.

File: pkg/workloads/k8s.go [hunk 2]
```diff
@@ -242,13 +242,18 @@ func TaskContainers(api *spec.API) ([]kcore.Container, []kcore.Volume) {
...
 				Name:      "neuron-sock",
 				MountPath: "/sock",
 			},
+			{
+				Name:      _kubexitGraveyardName,
+				MountPath: _kubexitGraveyardMountPath,
+			},
 		}
 		apiPodVolumeMounts = append(apiPodVolumeMounts, rtdVolumeMounts...)
-		neuronContainer := *neuronRuntimeDaemonContainer(api, rtdVolumeMounts)
```
[the remark this answers is not in the document]
Comment (miguelvr) [reply pkg/workloads/k8s.go:251]: our code base abuses the usage of pointers and it has created problems before.

there is no reason to return a pointer here. In this situation, the only reason to return a pointer would be if the function had the option of returning nil, which is not the case

File: pkg/workloads/k8s.go [hunk 3]
```diff
@@ -605,7 +640,7 @@ func tensorFlowHandlerContainers(api *spec.API, envVars []kcore.EnvVar, isJob bo
 		SecurityContext: &kcore.SecurityContext{
 			Privileged: pointer.Bool(true),
 		}},
-		*tensorflowServingContainer(
+		tensorflowServingContainer(
```
[the remark this answers is not in the document]
Comment (miguelvr) [reply pkg/workloads/k8s.go:643]: answered above

File: test/apis/batch/inferentia/cortex_inf.yaml (hunk 1 above)
Comment (miguelvr) [reply test/apis/batch/inferentia/cortex_inf.yaml:7]: this is under the `test` directory, so I don't think it is important to add a README. I can add the comment somewhere 👍

**Pull request merged**